cel/library.go

```diff
@@ -626,6 +620,9 @@ func (i *ipv4) decOctet() bool {
 //
 //	DIGIT = %x30-39  ; 0-9
 func (i *ipv4) digit() bool {
+	if i.index >= len(i.str) {
```

Comment:
The `digit()` helpers were inconsistent - the URI one had this guard in place but others didn't (and thus had to guard in each loop where they were used). This feels better to simplify the call sites.

Reply:
Agreed, consuming a DIGIT should simply return false if we're at the end of the input, and not require guards on call-sites.